docs/auth-modes.md

```diff
@@ -2,13 +2,13 @@
 title: Authentication Modes
```

Comment:
We'll need to rationalize this PR with https://github.com/tableau/connector-plugin-sdk/pull/860. Much of this content has been moved to the ui.md page